surround/assembler.py

```diff
@@ -0,0 +1,184 @@
+# assembler.py
+#
+# Manages a set of stages and the data that is passed between them.
+import sys
+import os
+import logging
+
+from abc import ABC
+from datetime import datetime
+
+from .config import Config
+from .loader import Loader
+from .surround import SurroundData
+from .visualiser import Visualiser
+from .stage import Validator, Filter, Estimator
+
+LOGGER = logging.getLogger(__name__)
+
+class Assembler(ABC):
+
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, assembler_name="", surround_data=None, estimator=None, config=None):
```

Comment:
We need some error handling on these parameters as the world blows up when I leave some of them out.

Reply:
I believe I check all of them before being used, that's why the initialisation is a bit open

Reply:
Yeah sure, I'll add some basic checking for this